README.md

```diff
@@ -4619,6 +4632,301 @@ The command above assumes you mounted the data workspace in `/mount/data`, and t
+
+# START HETEROGENEUS JOB 1
+
+read -r -d '' cmd_init_policy_inference <<EOF
+cd ${DIR} \
+&& export PYTHONPATH="${DIR}:${PYTHONPATH}" \
+&& export HYDRA_FULL_ERROR=1 \
+&& python rlhf/rlhf_nemo/serve_initial_policy.py \
+    --config-path=/opt/nemo-rlhf/rlhf/rlhf_nemo/conf \
+    --config-name=inference_initial_policy \
+    gpt_model_file=${ACTOR_MODEL} \
+    port=${INIT_POLICY_PORT=5556}
+EOF
+
+srun --het-group=1 -o $INIT_POLICY_OUTFILE -e $INIT_POLICY_ERRFILE --container-image=${CONTAINER} $MOUNTS bash -c "${cmd_init_policy_inference}" &
+
+# END HETEROGENEUS JOB 1
+
+sleep 30
+######################################################
+
+# START HETEROGENEUS JOB 2
+
+read -r -d '' cmd_critic_inference <<EOF
+cd ${DIR} \
+&& export PYTHONPATH="${DIR}:${PYTHONPATH}" \
+&& export HYDRA_FULL_ERROR=1 \
+&& python -u rlhf/rlhf_nemo/serve_ppo_critic.py \
+    --config-path=/opt/nemo-rlhf/rlhf/rlhf_nemo/conf \
+    --config-name=gpt_ppo_critic \
+    model.pretrained_checkpoint.restore_from_path=${RM_MODEL} \
+    inference.port=${CRITIC_PORT=5557}
+EOF
+
+srun --het-group=2 --container-image=${CONTAINER} bash -c "${cmd_critic_inference}" &
+
+# END HETEROGENEUS JOB 2
+
+sleep 30
+####################################################
+
+# START HETEROGENEUS JOB 3
+
+TRAIN_DATA_PATH=/path/to/train_data
+VALID_DATA_PATH=/path/to/val_data
+TEST_DATA_PATH=/path/to/test_data
+
+
+read -r -d '' cmd_ppo <<EOF
+cd ${DIR} \
+&& export PYTHONPATH="${DIR}:${PYTHONPATH}" \
+&& export HYDRA_FULL_ERROR=1 \
+&& python -u rlhf/rlhf_nemo/train_gpt_ppo_actor.py \
+    --config-path=/opt/nemo-rlhf/rlhf/rlhf_nemo/conf \
+    --config-name=gpt_ppo_actor \
+    trainer.num_nodes=8 \
+    "model.data.data_prefix={train: [${TRAIN_DATA_PATH}], validation: [${VALID_DATA_PATH}], test: [${TEST_DATA_PATH}]}" \
+    model.pretrained_checkpoint.restore_from_path=${ACTOR_MODEL} \
+    model.rlhf.reward_model.ip=${SLURM_JOB_NODELIST_HET_GROUP_0} \
+    model.rlhf.reward_model.port=${RM_PORT} \
+    model.rlhf.initial_policy.ip=${SLURM_JOB_NODELIST_HET_GROUP_1} \
+    model.rlhf.initial_policy.port=${INIT_POLICY_PORT} \
+    model.rlhf.critic.ip=${SLURM_JOB_NODELIST_HET_GROUP_2} \
+    model.rlhf.critic.port=${CRITIC_PORT}
+EOF
+
+srun --het-group=3 --container-image=${CONTAINER} bash -c "${cmd_ppo}" &
+
+# END HETEROGENEUS JOB 3
+
+wait
+```
+It is important to launch each job with & after the srun command, to ensure each job doesn’t block the next one. The wait statement at the end of script ensures that the entire job does not exit until each individual job is finished.
+
+Note: the three servers do not support data parallelism. Therefore, the SLURM –ntasks-per-node value should be set to the model parallelism value (tensor parallelism * pipeline parallelism) for that same job. And the trainer.devices value must also be set to that same value as well. However, the PPO actor supports data parallelism, so –ntasks-per-node can be set to the number of GPUs in each node.
```

Comment:
please highlight the `–ntasks-per-node`

Reply:
https://github.com/NVIDIA/NeMo-Megatron-Launcher/pull/30/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R4902